src/insertion_sort.py

```diff
@@ -0,0 +1,68 @@
+"""Python implementation of a Insertion Sort."""
+
+
+def insertion_sort(iter_item):
+    """Perform a insertion sort on an iterable."""
+    if type(iter_item) not in [list, tuple]:
+        raise TypeError('Please enter a list or tuple.')
+    sort_list = list(iter_item)
+    if type(sort_list[0]) not in [float, int]:
```

Comment:
Why are you doing this twice? I don't think you need to do it at all. It will still sort strings if you run it on strings.